File: Backend/AstroGame.Shared/Models/Technologies/TechnologyCost.cs
using AstroGame.Shared.Models.Resources;
using System;

namespace AstroGame.Shared.Models.Technologies
{
    public abstract class TechnologyCost
    {
        public Guid Id { get; set; }
        public Guid TechnologyId { get; set; }
        public Guid ResourceId { get; set; }

        public Technology Technology { get; set; }
        public Resource Resource { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/technologies/technology_cost.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/communications/converters/technology_cost_converter.dart';
import 'package:astrogame_app/models/common/guid.dart';

import 'package:json_annotation/json_annotation.dart';

@GuidConverter()
abstract class TechnologyCost {
  @JsonKey()
  late Guid id;

  @JsonKey()
  late Guid resourceId;

  @JsonKey()
  late Guid technologyId;

  TechnologyCost();

  factory TechnologyCost.fromJson(Map<String, dynamic> json) =>
      TechnologyCostConverter().fromJson(json);
  Map<String, dynamic> toJson() => TechnologyCostConverter().toJson(this);
}
